### PR TITLE
New headless support portal documentation

### PR DESCRIPTION
**Pull request by mattvagni**
Re-wrote this to better 'lead up' to building a support portal and assume less knowledge.

### CODE REVIEW

File: api-reference/headless-portal.mdx [hunk 1]
```diff
@@ -1,15 +1,54 @@
 ---
-title: 'Headless Portal'
+title: 'Headless Support Portal'
 description: 'Give your customers visibility over their support requests.'
 ---
 
-Customer support portals empower your customers to view, create and reply to support requests as well as share visibility of issues with their team.
+<Warning>
+Headless Portal support is only available in our custom pricing tier. If you are interested in using this feature please reach out to use on sales@plain.com.
```
Comment (jesushernandez) [inline api-reference/headless-portal.mdx:7]: `Scale` in plain.com/pricing (also what's used in the reply to thread docs)

File: api-reference/headless-portal.mdx [hunk 2]
```diff
@@ -1,15 +1,54 @@
...
+
+You must always make Plain API calls from an API you control. This allows you to implement access controls so that for example, a customer can only access their support requests etc.
+
+If you have leaked an API key by mistake, immediately delete it from your workspace settings and don't hesitate to reach out to us if we can help with any mitigation/investigations.
+
+### How does it work
+
+Details can vary depending on the experience you'd like to offer but in essence to build a portal you have 4 separate piece of work:
+
+1. Creating a tenant for each of your customers in Plain
```
Comment (jordrake) [inline api-reference/headless-portal.mdx:33]: I think we need some terminology here to seperate customer as a group i.e. Acme vs customer as an individual working for Acme. There's a mix of this in this document and it is quite confusing

File: api-reference/headless-portal.mdx [hunk 3]
```diff
@@ -22,41 +61,38 @@ To display a list of support requests you will need to fetch the appropriate thr
...
+To submit new support requests from your portal you can create a contact form which, when submitted, create a thread within Plain.
 
-## Authenticating
+What is important is that when you create the thread you pre-fill the customer's tenant id. This ensures that the support request they create is then visible in the support portal.
 
-In order to pass in the `tenantExternalId` or `customerCompanyId` shown above you will need to authenticate the user of the support portal. This is not shown in the example repository since it will be different for every application.
+You can read more on [how to create a thread here](/api-reference/graphql/threads/create).
 
-Once you have authenticated you may be able to get the id of their tenant directly from the claims, or you may need to query your backend systems.
+One thing to not is that you are not constrained at all here by the data you want to gather. Although a simple form with just a message and a title does work, it's typically much more useful and powerful to ask structured questions such as what the support request is about, how urgent it is and much more.
```
Comment (jesushernandez) [inline api-reference/headless-portal.mdx:78]:
```suggestion
One thing to note is that you are not constrained at all here by the data you want to gather. Although a simple form with just a message and a title does work, it's typically much more useful and powerful to ask structured questions such as what the support request is about, how urgent it is and much more.
```

`much more` could link to a page where we describe e.g. labels, priorities, etc?

File: api-reference/headless-portal.mdx [hunk 4]
```diff
@@ -22,41 +61,38 @@ To display a list of support requests you will need to fetch the appropriate thr
...
 
-You can allow customers to submit a new support request by [Creating a thread](/api-reference/graphql/threads/create). Be sure to set the tenant id when creating the thread if you're querying by tenant.
+</Step>
 
-## Viewing a request
+<Step title="Allowing customer to reply within a thread.">
+To reply to threads directly from your support portal is probably the most difficult part of building a good support portal from the UI side of things.
 
-In order for customers to view a support request you will want to provide the content by querying that thread's timeline. You can use the GraphQL API to query the exact combination of fields you need for your application, see the [API explorer](https://app.plain.com/developer/api-explorer/) for full details.
+In terms of the interaction with Plain however, it's as simple as calling the [`replyToThread` mutation](http://localhost:3000/api-reference/graphql/messaging/reply-to-thread).
```
Comment (jesushernandez) [inline api-reference/headless-portal.mdx:87]:
```suggestion
In terms of the interaction with Plain, however, it's as simple as calling the [`replyToThread` mutation](/api-reference/graphql/messaging/reply-to-thread).
```

File: api-reference/headless-portal.mdx (hunk 2 above)
Comment (mattvagni) [reply api-reference/headless-portal.mdx:35]: Have added an intro to point to our data model docs!

**Review by jordrake — APPROVED** [review]

**Review by jesushernandez — APPROVED** [review]

**Pull request merged**